lib/charms/grafana_k8s/v0/grafana_source.py

```diff
@@ -305,15 +315,14 @@ class GrafanaSourceEvents(ObjectEvents):
 class GrafanaSourceProvider(Object):
     """A provider object for Grafana datasources."""
 
-    _stored = StoredState()
-
     def __init__(
         self,
         charm: CharmBase,
+        source_type: str,
+        source_port: Optional[str] = "",
+        source_url: Optional[str] = "",
         refresh_event: Optional[BoundEvent] = None,
         relation_name: str = DEFAULT_RELATION_NAME,
-        source_type: Optional[str] = "prometheus",
-        source_port: Optional[str] = "9090",
```

Comment:
Is there a reason we are breaking backwards compatibility here by changing the defaults?

Reply:
Yes: they are bad defaults. They are not useful to anybody except Prometheus, and actively making it harder for people providing more datasources to understand how to use the API.